EvolutionModel/WateringHole.cs

```diff
@@ -25,20 +25,33 @@ public class WateringHole : IFoodContainer, IEatable {
 
     public bool CanBeEatenBy(Species eater) => !eater.IsPredator && FoodAmount > 0;
 
-        public bool GetsEaten(bool eater, out int foodEaten)
-        {
-            // Doe iets aan voorbereiding als dat nodig is.
-
-            if(CanBeEatenBy(eater)
-            {
+        public bool TryToEat(Species eater, out int amountEaten){
+            if(CanBeEatenBy(eater){
                 FoodAmount--;
-                foodEaten = 1;
+                amountEaten = 1;
                 return true;
             } 
             else {
-                foodEaten = default(int);
+                amountEaten = default(int);
                 return false;
             }
         }
+
+        public int GetsEaten(Species eater){
+            if(CanBeEatenBy(eater){
+                if(FoodAmount >= 1){
```

Comment:
Deze condition gaat nog steeds nooit false zijn, omdat hij ook in CanBeEatenBy false zou worden. Dan kan je hem net zo goed weghalen. Betekent wel dat de exceptiontekst algemener moet.